src/saveToCase.js

```diff
@@ -0,0 +1,475 @@
+define([
+    'vellum/form',
+    'jquery',
+    'underscore',
+    'vellum/mugs',
+    'vellum/parser',
+    'vellum/tree',
+    'vellum/util',
+    'vellum/widgets',
+    'tpl!vellum/templates/widget_update_case',
+    'vellum/core'
+], function (
+    form_,
+    $,
+    _,
+    mugs,
+    parser,
+    Tree,
+    util,
+    widgets,
+    widget_update_case
+){
+    function createsCase(mug) {
+        return mug ? mug.p.use_create : false;
+    }
+
+    function closesCase(mug) {
+        return mug ? mug.p.use_close : false;
+    }
+
+    function updatesCase(mug) {
+        return mug ? mug.p.use_update : false;
+    }
+
+    function addSetValue(mug) {
+        var path = mug.absolutePath;
+
+        if (createsCase(mug)) {
+            mug.form.addSetValue('xforms-ready', path + "/case/@case_id", mug.p.case_id);
+        }
+    }
+
+    var propertyWidget = function (mug, options) {
+            var widget = widgets.normal(mug, options),
+                id = options.id;
+
+            widget.kvInput = $('<div class="control-row" />').attr('name', id);
+
+            widget.getControl = function () {
+                return widget.kvInput;
+            };
```

Comment:
I used the same messages css class you used in `normal` so I don't think I need to define `widget.getMessagesContainer`. Would be nice to change it to show up at the correct property, but for now it's on the mug.